docs/smart-contracts/modules.md

```diff
@@ -0,0 +1,564 @@
+
+For example, you can implement the same governance for the **example** module using a governance capability named **GOVERNANCE** like this:
+
+```pact
+(module example GOVERNANCE
+  ...
+  (defcap GOVERNANCE ()
+    (enforce-keyset "admin-keyset"))
+  ...
+)
+```
+
+Note that the capability name has no significance, except to indicate the purpose of the capability. 
+Its placement at the beginning of the module declaration is what identifies this capability as a module governance capability.
+
+It's worth noting that, when you initially deploy a module, the module governance capability is not invoked. 
+This behavior is different than when you use use a keyset.
+With a keyset, the keyset must always be defined and evaluated to ensure that the keyset exists before a module can be deployed.
+Therefore, you might consider module governance using a capability to be more risky than using a keyset. 
+You should test modules thoroughly when implementing module governance capabilities to ensure that you don't introduce bugs that might prevent a module from being upgraded.
+
+### Invoking governance
+
+Because the module governance capability is defined using the `defcap` keyword, its administrative function cannot be called directly. 
+It is automatically invoked in the following circumstances:
+
+- Module upgrade is attempted.
+- Module tables are directly accessed from outside of the module code.
+
+Under these circumstances, the transaction is tested for elevated access to the **module administrator**, defined as the grant of the **module administrator capability**. 
+This capability cannot be expressed in user code, so it cannot be installed, acquired, required, or composed.
+
+However, the capability that implements the module administrator capability—in this example, the `GOVERNANCE` capability—can be installed, acquired, required, or composed. 
+If the `GOVERNANCE` capability is installed or granted, it's scoped like any other capability over the code that only the module administrator can run.
+
+### Module administrator scope
+
+The special module administrator capability, once automatically invoked, stays in scope for the rest of the calling transaction. 
+This is unlike other capabilities that can only be acquired in a fixed scope specified by the body of a `with-capability` function call.
+The reason for this difference in behavior is ensure that a governance capability doesn't rely on transient information that can change during a single transaction. 
+This is important, for example, in the case of module upgrades.
+A module upgrade might change the governance capability itself.
+If the module administrator didn't remain in scope for the completion of the transaction, the upgrade might fail, for example, because the upgrade requires migrating data with direct table rights.
```

Comment:
I dont understand this,  direct table rights?

Reply:
### Module admin capability scope

The special module admin capability, once automatically installed in the cases described above, stays in scope for the rest of the calling transaction. This is unlike "user" capabilities, which can only be acquired in a fixed scope specified by the body of with-capability.

This may sound worrisome, but the rationale is that a governance capability once granted should not be based on some transient fact that can become false during a single transaction. This is important especially in module upgrades, which can change the governance capability itself: if the module admin was tested again this could cause the upgrade to fail, for instance when migrating data with direct table rights.